core/src/impl/Kokkos_TaskQueueCommon.hpp

```diff
@@ -87,6 +87,7 @@ class TaskQueueCommonMixin {
   // <editor-fold desc="Constructors, destructor, and assignment"> {{{2
 
   TaskQueueCommonMixin() : m_ready_count(0) {
+    Kokkos::memory_fence();
```

Comment:
Did you mean to add this memory fence?

Reply:
yes

Reply:
I don't get why it is needed but not blocking